consensus/src/block_storage/sync_manager.rs

```diff
@@ -248,9 +249,42 @@ impl BlockStore {
         quorum_certs.extend(
             blocks
                 .iter()
-                .take(num_blocks as usize - 1)
+                .take(blocks.len() - 1)
                 .map(|block| block.quorum_cert().clone()),
         );
+
+        // check if highest_commit_cert is a fork,
+        // if so, we need to fetch it's block as well, to have a proof of commit.
+        if !blocks
+            .iter()
+            .any(|block| block.id() == highest_commit_cert.certified_block().id())
+        {
+            let fetched_commit_blocks = retriever
+                .retrieve_block_for_qc(
+                    highest_commit_cert,
+                    1,
+                    highest_commit_cert.commit_info().id(),
+                )
+                .await?;
+
+            assert_eq!(fetched_commit_blocks.len(), 1);
+            let block = fetched_commit_blocks
+                .into_iter()
+                .next()
+                .expect("blocks are empty");
+            assert_eq!(
+                block.id(),
+                highest_commit_cert.certified_block().id(),
```

Comment:
I don't think we need this, the block retrieval response verification should check it

Reply:
well, it could have not found it in the num_blocks asked, and not matched?

we do the same assert on the original fetch as well, I assume for the same reason.

Reply:
if it's not found or not matches the response will error out. (and we shouldn't panic on that since the behavior is uncontrolled)
the assertion is more like sanity check I guess